### DISCUSSION TITLE
Search for photos by date

### DISCUSSION THREAD

**Discussion by pelaxa** [immich-app/immich#3542] [Q&A]
Having a hard time finding how to search for photos by date.  I'd like to find photos taken between x and y dates through the API  so that I can display them on a [MagicMirror](https://magicmirror.builders/).

is this currently possible?

**Comment by jrasm91** [comment]
Not really. You could potentially use the two time bucket apis or the all asset API and then filter the response by date.

**Comment by pelaxa** [comment]
Thanks.  is this kind of search in the plans?

for now I think my best bet would be to use getMemoryLane in a loop for a few days that I am interested in, and then serveFile to display it on MagicMirror.

**Reply by jrasm91** [reply]
Yeah that's a good idea actually - showing memory lane photos on the mirror.

As far as search goes, yes there are plans to enhance it.

**Comment by avee87** [comment]
Even a simpler option of just typing a date in search would be very useful. I often search for something like "September 2021" to find photos from a specific period. It has a typeahead for dates leading to dedicated landing page that selects photos from the year/month in question.

**Comment by xa-bi** [comment]
It would be nice to search photos/videos by date and date ranges
Posible searches: `2003` `Jan 2014` `2008-12-21` `2004:2005` `2006-10-01:2006-10-20`...

**Comment by gitdoc67** [comment]
agreed, searching by date seems like it should be a core feature. To me its far more important than searching for an ML element.
